elasticsearch/helpers/vectorstore/_async/vectorstore.py

```diff
@@ -0,0 +1,384 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import logging
+import uuid
+from typing import Any, Callable, Dict, List, Optional
+
+from elasticsearch import AsyncElasticsearch
+from elasticsearch._version import __versionstr__ as lib_version
+from elasticsearch.helpers import BulkIndexError, async_bulk
+from elasticsearch.helpers.vectorstore import (
+    AsyncEmbeddingService,
+    AsyncRetrievalStrategy,
+)
+from elasticsearch.helpers.vectorstore._utils import maximal_marginal_relevance
+
+logger = logging.getLogger(__name__)
+
+
+class AsyncVectorStore:
+    """
+    VectorStore is a higher-level abstraction of indexing and search.
+    Users can pick from available retrieval strategies.
+
+    Documents have up to 3 fields:
+      - text_field: the text to be indexed and searched.
+      - metadata: additional information about the document, either schema-free
+        or defined by the supplied metadata_mappings.
+      - vector_field (usually not filled by the user): the embedding vector of the text.
+
+    Depending on the strategy, vector embeddings are
+      - created by the user beforehand
+      - created by this AsyncVectorStore class in Python
+      - created in-stack by inference pipelines.
+    """
+
+    def __init__(
+        self,
+        es_client: AsyncElasticsearch,
+        index_name: str,
```

Comment:
I believe everything that is not client should be keyword-only, again for consistency with the other APIs. This can be done by adding a `*` between, `client` and `index`.